internal/server/api_sources.go

```diff
@@ -127,3 +129,31 @@ func (s *Server) handleSourcesCreate(ctx context.Context, r *http.Request, body
 
 	return resp, nil
 }
+
+func (s *Server) handleSourcesDelete(ctx context.Context, r *http.Request, body []byte) (interface{}, *apiError) {
+	b64path := mux.Vars(r)["sourcePath"]
+
+	path, err := base64.URLEncoding.DecodeString(b64path)
```

Comment:
is there a reason for base64? wouldn't uri encode be sufficient?

Reply:
uri encode seems to cause the server to attempt to parse the string as part of the path, and respond accordingly, which can generate either no response (leading `/`) or a 404 not found error.

Reply:
as discussed on slack, can we switch to passing user/host/path as query string parameters and not URL path, just like `/sources/cancel`, `/sources/upload` and few others.